File: demo/macosApp/StellarDemo/Views/InvokeHelloWorldContractScreen.swift
import SwiftUI
import shared

// MARK: - Invoke Hello World Contract Screen (matches Compose InvokeHelloWorldContractScreen)

struct InvokeHelloWorldContractScreen: View {
    @ObservedObject var toastManager: ToastManager
    @State private var contractId = ""
    @State private var toParameter = ""
    @State private var submitterAccountId = ""
    @State private var secretKey = ""
    @State private var showSecret = false
    @State private var isInvoking = false
    @State private var invocationResult: InvokeHelloWorldResult?
    @State private var validationErrors: [String: String] = [:]

    var body: some View {
        ScrollView {
            VStack(spacing: 16) {
                infoCard
                contractDetailsCard
                submitterAccountCard
                invokeButton
                resultView
                placeholderView
            }
            .padding(16)
        }
        .background(Material3Colors.surface)
        .navigationToolbar(title: "Invoke Hello World Contract")
    }

    // MARK: - View Components

    private var infoCard: some View {
        InfoCard(title: "ContractClient.invoke(): Beginner-friendly contract invocation", color: .secondary) {
            Text("This demo showcases the SDK's high-level contract invocation API with automatic type conversion. The invoke() method accepts Map-based arguments and handles XDR conversion, transaction building, signing, submission, and result parsing automatically.")
                .font(.system(size: 13))
                .foregroundStyle(Material3Colors.onSecondaryContainer)
        }
    }

    private var contractDetailsCard: some View {
        InfoCard(title: "Contract Details", color: .default) {
            VStack(spacing: 12) {
                StellarTextField(
                    label: "Contract ID",
                    placeholder: "C...",
                    text: $contractId,
                    error: validationErrors["contractId"],
                    helpText: "Deploy hello world contract first using 'Deploy a Smart Contract'",
                    useMonospacedFont: true
                )
                .onChange(of: contractId) { _ in
                    validationErrors.removeValue(forKey: "contractId")
                    invocationResult = nil
                }

                StellarTextField(
                    label: "Name (to parameter)",
                    placeholder: "Alice",
                    text: $toParameter,
                    error: validationErrors["toParameter"],
                    helpText: "The name to greet in the hello function"
                )
                .onChange(of: toParameter) { _ in
                    validationErrors.removeValue(forKey: "toParameter")
                    invocationResult = nil
                }
            }
        }
    }

    private var submitterAccountCard: some View {
        InfoCard(title: "Submitter Account", color: .default) {
            VStack(spacing: 12) {
                StellarTextField(
                    label: "Submitter Account ID",
                    placeholder: "G...",
                    text: $submitterAccountId,
                    error: validationErrors["submitterAccount"],
                    helpText: "Account that will sign and submit the transaction",
                    useMonospacedFont: true
                )
                .onChange(of: submitterAccountId) { _ in
                    validationErrors.removeValue(forKey: "submitterAccount")
                    invocationResult = nil
                }

                StellarTextField(
                    label: "Secret Key",
                    placeholder: "S...",
                    text: $secretKey,
                    error: validationErrors["secretKey"],
                    isSecure: true,
                    showVisibilityToggle: true,
                    isVisible: $showSecret,
                    useMonospacedFont: true
                )
                .onChange(of: secretKey) { _ in
                    validationErrors.removeValue(forKey: "secretKey")
                    invocationResult = nil
                }
            }
        }
    }

    private var invokeButton: some View {
        LoadingButton(
            action: invokeContract,
            isLoading: isInvoking,
            isEnabled: !isInvoking && isFormValid,
            icon: "play.fill",
            text: "Invoke Contract",
            loadingText: "Invoking..."
        )
    }

    private var isFormValid: Bool {
        !contractId.isEmpty &&
        !toParameter.isEmpty &&
        !submitterAccountId.isEmpty &&
        !secretKey.isEmpty
    }

    @ViewBuilder
    private var resultView: some View {
        if let result = invocationResult {
            switch result {
            case let success as InvokeHelloWorldResult.Success:
                successCard(success)
            case let error as InvokeHelloWorldResult.Error:
                errorCard(error)
                troubleshootingCard
            default:
                EmptyView()
            }
        }
    }

    private func successCard(_ success: InvokeHelloWorldResult.Success) -> some View {
        VStack(spacing: 16) {
            // Success header card
            InfoCard(color: .success) {
                HStack(spacing: 8) {
                    Image(systemName: "checkmark.circle.fill")
                        .font(.system(size: 24))
                        .foregroundStyle(Material3Colors.onSuccessContainer)

                    VStack(alignment: .leading, spacing: 4) {
                        Text("Contract Invocation Successful")
                            .font(.system(size: 16, weight: .bold))
                            .foregroundStyle(Material3Colors.onSuccessContainer)

                        Text("The hello function was successfully invoked")
                            .font(.system(size: 14))
                            .foregroundStyle(Material3Colors.onSuccessContainer)
                    }
                }
            }

            // Greeting Response Card
            InfoCard(title: "Greeting Response", color: .surfaceVariant) {
                VStack(alignment: .leading, spacing: 12) {
                    Divider()

                    Text(success.greeting)
                        .font(.system(size: 20, weight: .medium, design: .default))
                        .foregroundStyle(Material3Colors.onSurfaceVariant)
                        .padding(16)
                        .frame(maxWidth: .infinity, alignment: .leading)
                        .background(Material3Colors.onSurfaceVariant.opacity(0.1))
                        .cornerRadius(8)
                }
            }

            // What's Next? Card
            InfoCard(title: "What's Next?", color: .secondary) {
                VStack(alignment: .leading, spacing: 4) {
                    Text("• The contract function was successfully invoked using ContractClient.invoke()")
                        .font(.system(size: 13))
                        .foregroundStyle(Material3Colors.onSecondaryContainer)

                    Text("• Automatic type conversion from Map arguments to Soroban XDR types")
                        .font(.system(size: 13))
                        .foregroundStyle(Material3Colors.onSecondaryContainer)

                    Text("• Try invoking with different names to see various greetings")
                        .font(.system(size: 13))
                        .foregroundStyle(Material3Colors.onSecondaryContainer)

                    Text("• View the transaction on Stellar Expert or other block explorers")
                        .font(.system(size: 13))
                        .foregroundStyle(Material3Colors.onSecondaryContainer)
                }
                .padding(.leading, 8)
            }
        }
    }

    private func errorCard(_ error: InvokeHelloWorldResult.Error) -> some View {
        InfoCard(color: .error) {
            HStack(spacing: 8) {
                Image(systemName: "exclamationmark.triangle.fill")
                    .foregroundStyle(Material3Colors.onErrorContainer)
                VStack(alignment: .leading, spacing: 8) {
                    Text("Invocation Failed")
                        .font(.system(size: 16, weight: .bold))
                        .foregroundStyle(Material3Colors.onErrorContainer)

                    Text(error.message)
                        .font(.system(size: 14))
                        .foregroundStyle(Material3Colors.onErrorContainer)

                    if let exception = error.exception {
                        Text("Technical details: \(exception.message ?? "Unknown error")")
                            .font(.system(size: 12, design: .monospaced))
                            .foregroundStyle(Material3Colors.onErrorContainer)
                    }
                }
            }
        }
    }

    private var troubleshootingCard: some View {
        InfoCard(title: "Troubleshooting", color: .secondary) {
            VStack(alignment: .leading, spacing: 4) {
                Text("• Ensure the contract ID is correct and the contract is deployed on testnet")
                    .font(.system(size: 13))
                    .foregroundStyle(Material3Colors.onSecondaryContainer)

                Text("• Verify the submitter account has sufficient XLM balance for fees")
                    .font(.system(size: 13))
                    .foregroundStyle(Material3Colors.onSecondaryContainer)

                Text("• Check that the secret key matches the submitter account ID")
                    .font(.system(size: 13))
                    .foregroundStyle(Material3Colors.onSecondaryContainer)

                Text("• Make sure you deployed the Hello World contract first (not another contract)")
                    .font(.system(size: 13))
                    .foregroundStyle(Material3Colors.onSecondaryContainer)

                Text("• Check your internet connection and Soroban RPC availability")
                    .font(.system(size: 13))
                    .foregroundStyle(Material3Colors.onSecondaryContainer)
            }
            .padding(.leading, 8)
        }
    }

    @ViewBuilder
    private var placeholderView: some View {
        if invocationResult == nil && !isInvoking && contractId.isEmpty {
            VStack(spacing: 16) {
                Image(systemName: "play.circle")
                    .font(.system(size: 64))
                    .foregroundStyle(Material3Colors.onSurfaceVariant.opacity(0.3))

                Text("Enter contract details to invoke the hello function")
                    .font(.system(size: 14))
                    .foregroundStyle(Material3Colors.onSurfaceVariant)
                    .multilineTextAlignment(.center)
            }
            .padding(.vertical, 32)
        }
    }

    // MARK: - Actions

    private func validateInputs() -> [String: String] {
        var errors: [String: String] = [:]

        // Use FormValidation for Stellar protocol fields
        if let error = FormValidation.validateContractIdField(contractId) {
            errors["contractId"] = error
        }

        // Domain-specific validation for "to" parameter (stays inline)
        if toParameter.isEmpty {
            errors["toParameter"] = "Name parameter is required"
        }

        if let error = FormValidation.validateAccountIdField(submitterAccountId) {
            errors["submitterAccount"] = error
        }

        if let error = FormValidation.validateSecretSeedField(secretKey) {
            errors["secretKey"] = error
        }

        return errors
    }

    private func invokeContract() {
        let errors = validateInputs()
        if !errors.isEmpty {
            validationErrors = errors
            return
        }

        isInvoking = true
        invocationResult = nil
        validationErrors = [:]

        Task {
            do {
                // Call the Kotlin function from the shared module
                let result = try await InvokeHelloWorldContractKt.invokeHelloWorldContract(
                    contractId: contractId,
                    to: toParameter,
                    submitterAccountId: submitterAccountId,
                    secretKey: secretKey,
                )

                await MainActor.run {
                    invocationResult = result
                    isInvoking = false
                }
            } catch {
                await MainActor.run {
                    invocationResult = InvokeHelloWorldResult.Error(
                        message: "Failed to invoke contract: \(error.localizedDescription)",
                        exception: error as? KotlinThrowable
                    )
                    isInvoking = false
                }
            }
        }
    }
}

File: demo/shared/src/commonMain/kotlin/com/soneso/demo/ui/screens/InvokeHelloWorldContractScreen.kt
package com.soneso.demo.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.text.KeyboardActions
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.text.selection.SelectionContainer
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Error
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.ImeAction
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import cafe.adriel.voyager.core.screen.Screen
import cafe.adriel.voyager.navigator.LocalNavigator
import cafe.adriel.voyager.navigator.currentOrThrow
import com.soneso.demo.stellar.InvokeHelloWorldResult
import com.soneso.demo.stellar.invokeHelloWorldContract
import com.soneso.demo.ui.FormValidation
import com.soneso.demo.ui.components.AnimatedButton
import com.soneso.demo.ui.components.InfoCardMediumTitle
import com.soneso.demo.ui.components.StellarTopBar
import com.soneso.demo.ui.components.*
import kotlinx.coroutines.launch

class InvokeHelloWorldContractScreen : Screen {
    @OptIn(ExperimentalMaterial3Api::class)
    @Composable
    override fun Content() {
        val navigator = LocalNavigator.currentOrThrow
        val coroutineScope = rememberCoroutineScope()

        // State management
        var contractId by remember { mutableStateOf("") }
        var toParameter by remember { mutableStateOf("") }
        var submitterAccountId by remember { mutableStateOf("") }
        var secretKey by remember { mutableStateOf("") }
        var isInvoking by remember { mutableStateOf(false) }
        var invocationResult by remember { mutableStateOf<InvokeHelloWorldResult?>(null) }
        var validationErrors by remember { mutableStateOf<Map<String, String>>(emptyMap()) }

        val snackbarHostState = remember { SnackbarHostState() }
        val scrollState = rememberScrollState()

        // Auto-scroll to bottom when invocation result appears
        LaunchedEffect(invocationResult) {
            invocationResult?.let {
                scrollState.animateScrollTo(scrollState.maxValue)
            }
        }

        // Validation function using FormValidation utility
        fun validateInputs(): Map<String, String> {
            val errors = mutableMapOf<String, String>()

            FormValidation.validateContractIdField(contractId)?.let {
                errors["contractId"] = it
            }

            if (toParameter.isBlank()) {
                errors["toParameter"] = "Name parameter is required"
            }

            FormValidation.validateAccountIdField(submitterAccountId)?.let {
                errors["submitterAccount"] = it
            }

            FormValidation.validateSecretSeedField(secretKey)?.let {
                errors["secretKey"] = it
            }

            return errors
        }

        // Function to invoke contract
        fun invokeContract() {
            val errors = validateInputs()
            if (errors.isNotEmpty()) {
                validationErrors = errors
                return
            }

            coroutineScope.launch {
                isInvoking = true
                invocationResult = null
                validationErrors = emptyMap()

                try {
                    invocationResult = invokeHelloWorldContract(
                        contractId = contractId,
                        to = toParameter,
                        submitterAccountId = submitterAccountId,
                        secretKey = secretKey,
                    )
                } finally {
                    isInvoking = false
                }
            }
        }

        Scaffold(
            topBar = {
                StellarTopBar(
                    title = "Invoke Hello World Contract",
                    onNavigationClick = { navigator.pop() }
                )
            },
            snackbarHost = { SnackbarHost(snackbarHostState) }
        ) { paddingValues ->
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(paddingValues),
                contentAlignment = Alignment.TopCenter
            ) {
                Column(
                    modifier = Modifier
                        .widthIn(max = 800.dp)
                        .fillMaxWidth()
                        .verticalScroll(scrollState)
                        .padding(20.dp),
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.spacedBy(20.dp)
                    ) {
                // Information card - Purple
                InfoCardMediumTitle(
                    modifier = Modifier.fillMaxWidth(),
                    title = "Beginner-friendly Contract Invocation",
                    description = "This demo showcases the SDK's high-level contract invocation API with automatic type conversion. " +
                            "The invoke() method accepts Map-based arguments and handles XDR conversion, transaction building, " +
                            "signing, submission, and result parsing automatically."
                )

                // Input fields card - Gold (consolidated)
                GoldCard(modifier = Modifier.fillMaxWidth()) {
                    Text(
                        text = "Contract Configuration",
                        style = MaterialTheme.typography.titleMedium.copy(
                            fontWeight = FontWeight.Bold,
                            lineHeight = 22.sp
                        ),
                        color = Color(0xFFA85A00) // Starlight Gold Dark
                    )

                    OutlinedTextField(
                        value = contractId,
                        onValueChange = {
                            contractId = it.trim()
                            validationErrors = validationErrors - "contractId"
                            invocationResult = null
                        },
                        label = { Text("Contract ID") },
                        placeholder = { Text("C...") },
                        modifier = Modifier.fillMaxWidth(),
                        singleLine = true,
                        isError = validationErrors.containsKey("contractId"),
                        supportingText = validationErrors["contractId"]?.let { error ->
                            {
                                Text(
                                    text = error,
                                    color = Color(0xFF991B1B)
                                )
                            }
                        } ?: {
                            Text(
                                text = "Deploy hello world contract first using 'Deploy a Smart Contract'",
                                color = Color(0xFFA85A00).copy(alpha = 0.7f)
                            )
                        },
                        colors = OutlinedTextFieldDefaults.colors(
                            focusedBorderColor = Color(0xFFD97706),
                            focusedLabelColor = Color(0xFFD97706),
                            cursorColor = Color(0xFFD97706)
                        ),
                        keyboardOptions = KeyboardOptions(
                            imeAction = ImeAction.Next
                        )
                    )

                    OutlinedTextField(
                        value = toParameter,
                        onValueChange = {
                            toParameter = it
                            validationErrors = validationErrors - "toParameter"
                            invocationResult = null
                        },
                        label = { Text("Name (to parameter)") },
                        placeholder = { Text("Alice") },
                        modifier = Modifier.fillMaxWidth(),
                        singleLine = true,
                        isError = validationErrors.containsKey("toParameter"),
                        supportingText = validationErrors["toParameter"]?.let { error ->
                            {
                                Text(
                                    text = error,
                                    color = Color(0xFF991B1B)
                                )
                            }
                        } ?: {
                            Text(
                                text = "The name to greet in the hello function",
                                color = Color(0xFFA85A00).copy(alpha = 0.7f)
                            )
                        },
                        colors = OutlinedTextFieldDefaults.colors(
                            focusedBorderColor = Color(0xFFD97706),
                            focusedLabelColor = Color(0xFFD97706),
                            cursorColor = Color(0xFFD97706)
                        ),
                        keyboardOptions = KeyboardOptions(
                            imeAction = ImeAction.Next
                        )
                    )

                    HorizontalDivider(color = Color(0xFFD97706).copy(alpha = 0.2f))

                    Text(
                        text = "Submitter Account",
                        style = MaterialTheme.typography.titleSmall.copy(
                            fontWeight = FontWeight.Bold
                        ),
                        color = Color(0xFFA85A00)
                    )

                    OutlinedTextField(
                        value = submitterAccountId,
                        onValueChange = {
                            submitterAccountId = it.trim()
                            validationErrors = validationErrors - "submitterAccount"
                            invocationResult = null
                        },
                        label = { Text("Submitter Account ID") },
                        placeholder = { Text("G...") },
                        modifier = Modifier.fillMaxWidth(),
                        singleLine = true,
                        isError = validationErrors.containsKey("submitterAccount"),
                        supportingText = validationErrors["submitterAccount"]?.let { error ->
                            {
                                Text(
                                    text = error,
                                    color = Color(0xFF991B1B)
                                )
                            }
                        } ?: {
                            Text(
                                text = "Account that will sign and submit the transaction",
                                color = Color(0xFFA85A00).copy(alpha = 0.7f)
                            )
                        },
                        colors = OutlinedTextFieldDefaults.colors(
                            focusedBorderColor = Color(0xFFD97706),
                            focusedLabelColor = Color(0xFFD97706),
                            cursorColor = Color(0xFFD97706)
                        ),
                        keyboardOptions = KeyboardOptions(
                            imeAction = ImeAction.Next
                        )
                    )

                    OutlinedTextField(
                        value = secretKey,
                        onValueChange = {
                            secretKey = it.trim()
                            validationErrors = validationErrors - "secretKey"
                            invocationResult = null
                        },
                        label = { Text("Secret Key") },
                        placeholder = { Text("S...") },
                        visualTransformation = PasswordVisualTransformation(),
                        modifier = Modifier.fillMaxWidth(),
                        singleLine = true,
                        isError = validationErrors.containsKey("secretKey"),
                        supportingText = validationErrors["secretKey"]?.let { error ->
                            {
                                Text(
                                    text = error,
                                    color = Color(0xFF991B1B)
                                )
                            }
                        },
                        colors = OutlinedTextFieldDefaults.colors(
                            focusedBorderColor = Color(0xFFD97706),
                            focusedLabelColor = Color(0xFFD97706),
                            cursorColor = Color(0xFFD97706)
                        ),
                        keyboardOptions = KeyboardOptions(
                            imeAction = ImeAction.Done
                        ),
                        keyboardActions = KeyboardActions(
                            onDone = { invokeContract() }
                        )
                    )
                }

                // Invoke button with AnimatedButton
                AnimatedButton(
                    onClick = { invokeContract() },
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(56.dp),
                    enabled = contractId.isNotBlank() &&
                            toParameter.isNotBlank() &&
                            submitterAccountId.isNotBlank() &&
                            secretKey.isNotBlank(),
                    isLoading = isInvoking,
                    colors = ButtonDefaults.buttonColors(
                        containerColor = Color(0xFF0A4FD6), // Stellar Blue
                        contentColor = Color.White
                    )
                ) {
                    Icon(
                        imageVector = Icons.Default.PlayArrow,
                        contentDescription = null,
                        tint = Color.White
                    )
                    Spacer(modifier = Modifier.width(8.dp))
                    Text(
                        "Invoke Contract",
                        style = MaterialTheme.typography.bodyLarge.copy(
                            fontWeight = FontWeight.Bold
                        )
                    )
                }

                // Result display
                invocationResult?.let { result ->
                    when (result) {
                        is InvokeHelloWorldResult.Success -> {
                            SuccessCard(result)
                        }
                        is InvokeHelloWorldResult.Error -> {
                            com.soneso.demo.ui.components.ErrorCard(modifier = Modifier.fillMaxWidth()) {
                                Row(
                                    verticalAlignment = Alignment.CenterVertically,
                                    horizontalArrangement = Arrangement.spacedBy(12.dp)
                                ) {
                                    Icon(
                                        imageVector = Icons.Default.Error,
                                        contentDescription = null,
                                        tint = Color(0xFF991B1B), // Nova Red Dark
                                        modifier = Modifier.size(28.dp)
                                    )
                                    Text(
                                        text = "Error",
                                        style = MaterialTheme.typography.titleLarge.copy(
                                            fontWeight = FontWeight.Bold,
                                            lineHeight = 22.sp
                                        ),
                                        color = Color(0xFF991B1B)
                                    )
                                }

                                Text(
                                    text = result.message,
                                    style = MaterialTheme.typography.bodyMedium.copy(
                                        fontWeight = FontWeight.SemiBold,
                                        lineHeight = 22.sp
                                    ),
                                    color = Color(0xFF991B1B).copy(alpha = 0.9f)
                                )

                                // Technical details in white nested card
                                result.exception?.let { exception ->
                                    Card(
                                        modifier = Modifier.fillMaxWidth(),
                                        colors = CardDefaults.cardColors(
                                            containerColor = Color.White
                                        )
                                    ) {
                                        SelectionContainer {
                                            Column(
                                                modifier = Modifier.padding(16.dp),
                                                verticalArrangement = Arrangement.spacedBy(8.dp)
                                            ) {
                                                Text(
                                                    text = "Technical Details",
                                                    style = MaterialTheme.typography.labelMedium.copy(
                                                        fontWeight = FontWeight.Bold
                                                    ),
                                                    color = Color(0xFF991B1B).copy(alpha = 0.7f)
                                                )
                                                Text(
                                                    text = exception.message ?: "Unknown error",
                                                    style = MaterialTheme.typography.bodySmall.copy(
                                                        fontFamily = FontFamily.Monospace,
                                                        lineHeight = 20.sp
                                                    ),
                                                    color = Color(0xFF991B1B).copy(alpha = 0.8f)
                                                )
                                            }
                                        }
                                    }
                                }
                            }

                            Spacer(modifier = Modifier.height(12.dp))

                            // Purple troubleshooting tips card - ONLY ON ERROR
                            InfoCardMediumTitle(
                                modifier = Modifier.fillMaxWidth(),
                                title = "Troubleshooting",
                                description = "• Ensure the contract ID is correct and the contract is deployed on testnet\n" +
                                        "• Verify the submitter account has sufficient XLM balance for fees\n" +
                                        "• Check that the secret key matches the submitter account ID\n" +
                                        "• Make sure you deployed the Hello World contract first (not another contract)\n" +
                                        "• Check your internet connection and Soroban RPC availability"
                            )
                        }
                    }
                }
            }
        }
    }
}

@Composable
private fun SuccessCard(result: InvokeHelloWorldResult.Success) {
    // Teal success header
    TealCard(modifier = Modifier.fillMaxWidth()) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Icon(
                imageVector = Icons.Default.Check,
                contentDescription = null,
                tint = Color(0xFF0F766E), // Nebula Teal Dark
                modifier = Modifier.size(28.dp)
            )
            Text(
                text = "Success",
                style = MaterialTheme.typography.titleLarge.copy(
                    fontWeight = FontWeight.Bold,
                    lineHeight = 22.sp
                ),
                color = Color(0xFF0F766E)
            )
        }
    }

    // Blue greeting response card
    BlueCard(modifier = Modifier.fillMaxWidth()) {
        Text(
            text = "Greeting Response",
            style = MaterialTheme.typography.titleMedium.copy(
                fontWeight = FontWeight.Bold,
                lineHeight = 22.sp
            ),
            color = Color(0xFF0639A3) // Stellar Blue Dark
        )

        SelectionContainer {
            Card(
                modifier = Modifier.fillMaxWidth(),
                colors = CardDefaults.cardColors(
                    containerColor = Color.White
                )
            ) {
                Text(
                    text = result.greeting,
                    style = MaterialTheme.typography.titleLarge.copy(
                        fontWeight = FontWeight.Medium,
                        lineHeight = 28.sp
                    ),
                    color = Color(0xFF0639A3),
                    modifier = Modifier.padding(16.dp)
                )
            }
        }

        HorizontalDivider(color = Color(0xFF0639A3).copy(alpha = 0.2f))

        Text(
            text = "The contract function was successfully invoked using ContractClient.invoke() with automatic type conversion from Map arguments to Soroban XDR types.",
            style = MaterialTheme.typography.bodySmall.copy(
                fontWeight = FontWeight.SemiBold,
                lineHeight = 22.sp
            ),
            color = Color(0xFF0639A3).copy(alpha = 0.8f)
        )
    }
}
}
